lib/mix/test/mix/tasks/escript_test.exs

```diff
@@ -97,6 +108,19 @@ defmodule Mix.Tasks.EscriptTest do
     end
   end
 
+  test "generate escript with debug information" do
```

Comment:
We need to actually check in the test that the debug information has been stripped but I am not quite sure how to do that besides opening up the escript zip file and checking the beam contents. :(